src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -297,6 +302,27 @@ public RevisionsCollectionConfiguration GetRevisionsConfiguration(string collect
                     }
                 }
 
+                if (configuration.IncludeCountersSnapshot &&
+                    document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) &&
+                    metadata.TryGet(Constants.Documents.Metadata.Counters, out BlittableJsonReaderArray counterNames))
+                {
+                    var countersSnapshot = counterNames.ToDictionary(c => c.ToString(), 
```

Comment:
Let's avoid having complex nested Linq ops.
Build the `DynamicJsonObject` directly

Reply:
done